### PR TITLE
Milestone6: Preserve data

### PR DESCRIPTION
**Pull request by Favourezeugwa**
## Implemented the following in this milestone;
-  Saved all the data (books, people, and rentals) into JSON files (`rentals.json`, `person.json`, `book.json`) On exit from the app
- On startup of the app,  loaded all data from files.

### CODE REVIEW

File: JSONdata/handle_json.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      indent: '  ',
+      space_before: ' ',
+      space: ' '
+    }
+    File.write(file_path, JSON.generate(array, opts))
+  end
+
+  # READER
+  def read_json(file_path)
+    file = File.read(file_path)
```
Comment (omarsalem7) [inline JSONdata/handle_json.rb:20]: - [ ] Consider handling the case when one or more of the files are missing by using the `exists?` method (the app should work and show no errors).

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @Favourezeugwa 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- There is no linter error present ✔️
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @Favourezeugwa 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Favourezeugwa** [comment]
Thank you @omarsalem7 😀

**Pull request merged**